src/components/section/AllProducts.js

```diff
@@ -0,0 +1,62 @@
+import { fetchProducts } from "../../api/product";
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import heartIcon from "../../assets/icons/heart-icon.svg";
+import defaultImg from "../../assets/images/item-default-img-sm.svg";
+import "./AllProducts.css";
+
+function AllProducts({ sortOption }) {
+  const [items, setItems] = useState([]);
+  const [pageSize, setPageSize] = useState(10);
+
+  const updatePageSize = () => {
+    const width = window.innerWidth;
+    if (width > 1199) {
+      setPageSize(10);
+    } else if (width > 768) {
+      setPageSize(6);
+    } else {
+      setPageSize(4);
+    }
+  };
+
+  const loadProducts = async () => {
+    const { list } = await fetchProducts(sortOption, pageSize);
+    setItems(list);
+  };
+
+  useEffect(() => {
+    loadProducts();
+    updatePageSize();
+    window.addEventListener("resize", updatePageSize);
+    return () => window.removeEventListener("resize", updatePageSize);
+  }, [sortOption, pageSize]);
+
+  return (
+    <section className="all-products">
+      <div className="all-products-list">
+        {items.map((item) => {
+          return (
```

Comment:
딱히 리턴함에 있어서 특별한 로직 처리가 필요 없다면 다음과 같이 해줘도 무난합니다
```suggestion
        {items.map((item) => <Link ...
```